service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/GuardianMapper.java

```diff
@@ -0,0 +1,37 @@
+package uk.nhs.adaptors.oneoneone.cda.report.mapper;
+
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.Reference;
+
+import lombok.AllArgsConstructor;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Guardian;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Person;
+
+import org.hl7.fhir.dstu3.model.Patient;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class GuardianMapper {
+
+    private AddressMapper addressMapper;
+    private HumanNameMapper humanNameMapper;
+    private OrganizationMapper organizationMapper;
+
+    public Patient.ContactComponent mapGuardian(POCDMT000002UK01Guardian guardian) {
+        Patient.ContactComponent contactComponent = new Patient.ContactComponent();
+        if (guardian.isSetGuardianPerson()) {
+            POCDMT000002UK01Person guardianPerson = guardian.getGuardianPerson();
+            contactComponent.setName(humanNameMapper.mapHumanName(guardianPerson.getNameArray(0)));
```

Comment:
i would check if there is NameArray set using sizeOfNameArray >0

Reply:
Checked the size of the NameArray > 0 before processing.